src/listpack.c

```diff
@@ -1112,6 +1112,45 @@ unsigned char *lpBatchDelete(unsigned char *lp, unsigned char **ps, unsigned lon
     return lpShrinkToFit(lp);
 }
 
+/* Trim a listpack, remove 'ltrim' elements from left and 'rtrim' elements from right. */
+unsigned char *lpTrim(unsigned char *lp, unsigned long ltrim, unsigned long rtrim) {
+    if (ltrim == 0 && rtrim == 0) return lp;
+    size_t bytes = lpBytes(lp);
+    unsigned long llen = lpLength(lp);
+    if (ltrim + rtrim >= llen) {
+        /* Remove all elements */
+        lp[LP_HDR_SIZE] = LP_EOF;
+        lpSetTotalBytes(lp, LP_HDR_SIZE + 1);
+        lpSetNumElements(lp, 0);
+        return lpShrinkToFit(lp);
+    }
+    unsigned long rangelen = llen - ltrim - rtrim;
+    unsigned char *first, *tail; /* first points to the first byte we want to save, 
+                                    and tail points to the byte after the last byte we want to save */
+    first = lpSeek(lp, ltrim);
+    tail = lp + bytes - 1; /* When rtrim is zero, tail points to the EOF char */
+    if (rtrim) {
+        if (rangelen < rtrim) {
+            tail = first;
+            long num = rangelen;
+            while (num--) {
+                tail = lpSkip(tail);
+            }
+            assert(tail[0] != LP_EOF);
+            lpAssertValidEntry(lp, bytes, tail);
+        } else {
+            tail = lpSeek(lp, llen - (int)rtrim);
+        }
+    }
+    size_t rangeBytes = tail - first;
+    memmove(lp + LP_HDR_SIZE, first, rangeBytes);
+    lp[LP_HDR_SIZE + rangeBytes] = LP_EOF;
+    lpSetTotalBytes(lp, LP_HDR_SIZE + rangeBytes + 1);
+    lpSetNumElements(lp, rangelen);
+    lp = lpShrinkToFit(lp);
+    return lp;
```

Comment:
nit:

Can be symmetrical to the above if block.

```suggestion
    return lpShrinkToFit(lp);
```